packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -370,7 +370,11 @@ export async function handleRoute({
 	// Apply the `status` override to the response object before responding.
 	// Response.status is read-only, so a clone is required to override.
 	if (status && response.status !== status && (status === 404 || status === 500)) {
-		response = new Response(response.body, { ...response, status });
+		response = new Response(response.body, {
+			status: status,
+			statusText: status === 404 ? 'Not Found' : 'Internal Server Error',
+			headers: response.headers
+		});
```

Comment:
Makes sense why `...response` has no effect, spreading only gives you enumerable properties, `"headers"` (and all others) are instance properties instead.

```suggestion
		response = new Response(response.body, {
			status: status,
			headers: response.headers
		});
```

This is minor and a little subjective, but we would want to avoid setting `statusText`. It is not part of http 2, so it will be error prone if users depend on it being a certain value.

Reply:
Good point!